### PR TITLE
Add resource provider usages API for placement

### PR DESCRIPTION
**Pull request by dkt26111**
For #526

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

API doc:
https://docs.openstack.org/api-ref/placement/#list-resource-provider-usages

API code:
https://github.com/openstack/placement/blob/master/placement/handlers/usage.py#L38

### CODE REVIEW

**Comment by dkt26111** [comment]
@jtopjian This is ready for review.

File: openstack/placement/v1/resourceproviders/results.go [hunk 1]
```diff
@@ -33,6 +33,17 @@ type ResourceProvider struct {
...
 
+type Usage struct {
+	DiskGB   int `json:"DISK_GB"`
+	MemoryMB int `json:"MEMORY_MB"`
+	VCPU     int `json:"VCPU"`
+}
+
+type ResourceProviderUsage struct {
+	ResourceProviderGeneration int   `json:"resource_provider_generation"`
+	Usages                     Usage `json:"usages"`
```
Comment (jtopjian) [inline openstack/placement/v1/resourceproviders/results.go]: I'm thinking this should be `map[string]int`? 

I don't see any reference to `DISK_GB` etc in the placement code and the API docs mention:

>  The value is a dictionary of resource classes paired with the sum of the allocations of that resource class for this resource provider.

Given it's possible to [create arbitrary resource classes](https://docs.openstack.org/api-ref/placement/#create-resource-class), I think the result of `Usages` is going to be highly dynamic.

Thoughts?

Comment (dkt26111) [reply openstack/placement/v1/resourceproviders/results.go]: Sounds good to me.  I'll make the change.

**Comment by jtopjian** [comment]
Implemented in #1862

**Pull request closed without merging**